packages/block-editor/src/store/actions.js

```diff
@@ -248,61 +300,139 @@ export const moveBlocksDown = createOnMove( 'MOVE_BLOCKS_DOWN' );
  * inserted, optionally at a specific index respective a root block list.
  *
- * @param {Object}  block            Block object to insert.
- * @param {?number} index            Index at which block should be inserted.
- * @param {?string} rootClientId     Optional root client ID of block list on which to insert.
- * @param {?boolean} updateSelection If true block selection will be updated. If false, block selection will not change. Defaults to true.
+ * @param {Object}   block                         Block object to insert.
+ * @param {?number}  index                         Index at which block should be inserted.
+ * @param {?string}  rootClientId                  Optional root client ID of block list on which to insert.
+ * @param {?boolean} updateSelection               If true block selection will be updated. If false, block selection will not change. Defaults to true.
+ * @param {?boolean} ignoreAllowedBlocksValidation If true the block will be inserted even if the insertion was not allowed e.g: because of allowed blocks restrictions.
  *
  * @return {Object} Action object.
  */
-export function insertBlock( block, index, rootClientId, updateSelection = true ) {
-	return insertBlocks( [ block ], index, rootClientId, updateSelection );
+export function insertBlock(
+	block,
+	index,
+	rootClientId,
+	updateSelection = true,
+	ignoreAllowedBlocksValidation = false
+) {
+	return insertBlocks(
+		[ block ],
+		index,
+		rootClientId,
+		updateSelection,
+		ignoreAllowedBlocksValidation
+	);
 }
 
 /**
  * Returns an action object used in signalling that an array of blocks should
  * be inserted, optionally at a specific index respective a root block list.
  *
- * @param {Object[]} blocks          Block objects to insert.
- * @param {?number}  index           Index at which block should be inserted.
- * @param {?string}  rootClientId    Optional root client ID of block list on which to insert.
- * @param {?boolean} updateSelection If true block selection will be updated.  If false, block selection will not change. Defaults to true.
- *
- * @return {Object} Action object.
- */
-export function insertBlocks( blocks, index, rootClientId, updateSelection = true ) {
-	return {
-		type: 'INSERT_BLOCKS',
-		blocks: castArray( blocks ),
-		index,
-		rootClientId,
-		time: Date.now(),
-		updateSelection,
-	};
+ * @param {Object[]} blocks                        Block objects to insert.
+ * @param {?number}  index                         Index at which block should be inserted.
+ * @param {?string}  rootClientId                  Optional root client ID of block list on which to insert.
+ * @param {?boolean} updateSelection               If true block selection will be updated.  If false, block selection will not change. Defaults to true.
+ * @param {?boolean} ignoreAllowedBlocksValidation If true the block will be inserted even if the insertion was not allowed e.g: because of allowed blocks restrictions.
+ */
+export function* insertBlocks(
```

Comment:
It might be easier to introduce a new action `replaceInnerBlocks` which will skip validation by duplicating some logic from the existing methods rather than increasing the complexity of the existing actions by introducing another boolean param. It also seems like this new param is a special case required only to keep inner blocks working as before.

Reply:
> It might be easier to introduce a new action `replaceInnerBlocks` which will skip validation by duplicating some logic from the existing methods rather than increasing the complexity of the existing actions by introducing another boolean param. It also seems like this new param is a special case required only to keep inner blocks working as before.

Had you considered this feedback, @jorgefilipecosta ?

Reply:
Hi @aduth yes I followed this suggestion and the new action was already merged.